waiter/container_delete.go

```diff
@@ -0,0 +1,66 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	"github.com/nspcc-dev/neofs-sdk-go/container"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+)
+
+// ContainerDeleteExecutor represents requirements to async container delete operation.
+// See documentation for functions in [client.Client]. The same semantics is expected.
+type ContainerDeleteExecutor interface {
+	ContainerDelete(ctx context.Context, id cid.ID, prm client.PrmContainerDelete) error
+	ContainerGet(ctx context.Context, id cid.ID, prm client.PrmContainerGet) (container.Container, error)
+}
+
+// ContainerDeleteWaiter implements sync logic to container delete operation.
+type ContainerDeleteWaiter struct {
+	executor     ContainerDeleteExecutor
+	pollInterval time.Duration
+}
+
+// NewContainerDeleteWaiter is a constructor for ContainerDeleteWaiter.
+func NewContainerDeleteWaiter(executor ContainerDeleteExecutor, pollInterval time.Duration) ContainerDeleteWaiter {
+	return ContainerDeleteWaiter{executor: executor, pollInterval: pollInterval}
+}
+
+// SetPollInterval allows rewrite default poll interval.
+func (w *ContainerDeleteWaiter) SetPollInterval(interval time.Duration) {
+	w.pollInterval = interval
+}
+
+// ContainerDelete sends request to remove the NeoFS container.
+//
+// See documentation for function in [client.Client.ContainerDelete].
+func (w ContainerDeleteWaiter) ContainerDelete(ctx context.Context, id cid.ID, prm client.PrmContainerDelete) error {
+	if err := w.executor.ContainerDelete(ctx, id, prm); err != nil {
+		return fmt.Errorf("delete: %w", err)
+	}
+
+	var prmGet client.PrmContainerGet
+
+	logic := func() (bool, error) {
+		_, err := w.executor.ContainerGet(ctx, id, prmGet)
+		if err != nil {
+			if errors.Is(err, apistatus.ErrContainerNotFound) {
+				return true, nil
+			}
+
+			return false, fmt.Errorf("ContainerGet: %w", err)
+		}
+
+		return false, nil
+	}
+
+	if err := poll(ctx, w.pollInterval, logic); err != nil {
+		return fmt.Errorf("poll: %w", err)
```

Comment:
...and then adding another one here. Likely one (or none) is enough and this could be a `return poll()`.

Reply:
Replaced by `return poll(...`